scapy/contrib/bgp_fields.py

```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""BGP-4 disector: fields for implementing MPBGP extensions"""
+
+import socket
+if not socket.has_ipv6:
+    raise socket.error("can't use AF_INET6, IPv6 is disabled")
```

Comment:
Depending on what you are trying to parse, you might be able to know how to decode AS using OPEN messages. I think that the best solution will be to add a "conf.contrib.bgp.use_asn4".